OpenRA.Game/Map/CellLayer.cs

```diff
@@ -18,13 +18,13 @@ namespace OpenRA
 	// Represents a layer of "something" that covers the map
 	public class CellLayer<T> : IEnumerable<T>
 	{
-		public readonly Size Size;
-		public readonly TileShape Shape;
+		public Size Size { get; private set; }
```

Comment:
There's a workaround to keep the public readonly fields while providing interfaces. For the time being I'll switch back to readonly public properties, but consider moving to public private set properties for those cases in hopes of a better testable code.